### PR TITLE
Blacken entire project

### PR DESCRIPTION
**Pull request by baurt**
Blacken the entire project.  Solves https://github.com/jwag956/flask-security/issues/87.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -6,88 +6,80 @@
...
+    "psycopg2>=2.7.4",
+    "pydocstyle>=1.0.0",
+    "pymysql>=0.9.3",
+    "pyqrcode>=1.2",
+    "pytest-black>=0.3.7",
+    "pytest-cache>=1.0",
+    "pytest-cov>=2.5.1",
+    "pytest-flakes>=3.0.2",
+    "pytest-mongo>=1.2.1",
+    "pytest-pep8>=1.0.6",
```
Comment (jwag956) [inline setup.py]: do we still need this?

Comment (baurt) [reply setup.py]: We can get rid of that one.

**Comment by baurt** [comment]
Black shouldn't be part of tests require. The CI tests would fail for python less than 3.6. pytest-black actually skips the formatting checks if black is not available. The only thing you can modify is the line length.

I completely forgot about linting though. I'm thinking about flake8 or pylint (for things like unused imports.) I'm going to do another commit, but it gets tricky with the long key:value pairs(two factor change method password conformation in core.py) 

To actually use this, the only thing that needs to be done is run black on the files. Than run linting, fix errors and ignore lines if necessary.

**Comment by baurt** [comment]
Maybe some precommit hooks should be added in order to force linting and such

**Comment by jwag956** [comment]
Thanks - so what precisely do I (and other contributors) need  to do? I.e. could you at least in the commit explain/point to links that explain how to conform? Maybe this is the first paragraph of a 'developers' RST documentation???

flakes is still part of the pytest/setup line - what else are you thinking? We can always improve - just don't want to backslide.

**Comment by jwag956** [comment]
I dont think we should add a precommit hook - that basically forces people to do development a particular way - we already have PRs that run CI - so that will catch anything we need.

We have pytest-flakes - so that is a basic linter and since your CI ran succeeded it means you managed to get things right.
As for names like 'two factor change method password conformation' that is simply painful - we should change the name - it isn't really 'change_method' anyway it for changing anything in your 2FA profile (and I am working on opt-in mode). So we could just change that to tf_confirm_password ...

**Comment by baurt** [comment]
I'll work on a CONTRIBUTING.rst for this. Not much is required, except running black on the code and making sure that flake8 isn't throwing errors.

pytest-flakes really only catches logical errors, not things like unused imports.  flake8 incorporates pyflakes (pytest-flakes is based off of this) and pycodestyle (formerly pep8, which is what pytest-pep8 is based on).

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+The code is hosted on a GitHub repo at
+https://github.com/jwag956/flask-security.  To get a working environment, follow
+these steps:
+
+#.  (Optional, but recommended) Create a Python 3.6 (or greater) virtualenv to work in,
+    and activate it.  `Poetry <https://github.com/sdispater/poetry>`_ is a great tool to
+    automate installation of virtualenvs and requirements.
+
+.. like this:
+ mkvirtualenv -p /usr/local/pythonz/pythons/CPython-2.7.11/bin/python coverage
```
Comment (jwag956) [inline CONTRIBUTING.rst]: bad example? python 2.7?

Comment (baurt) [reply CONTRIBUTING.rst]: That is a bad example, I'm probably going to remove that.  Just an artifact from a contributing.rst (coverage.py) that I found.

**Comment by baurt** [comment]
Fair enough.  Removed the Contributing docs for now.  I'm still looking at Poetry for building.  Currently, I just use it for the virtualenv management and dependency resolution.

**Comment by jwag956** [comment]
closes: #87

**Pull request merged**